src/ui/TableOfContent.tsx

```diff
@@ -19,60 +77,57 @@ export default function TableOfContent({
   const tocLinkHandler = (evt: any) => {
     evt.preventDefault();
     navigator.goToPage(evt.target.getAttribute('href'));
-    setIsOpen(!isOpen);
+    setIsOpen(false);
   };
 
+  const tocBgColor = useColorModeValue('ui.white', 'ui.black', 'ui.sepia');
+
   return (
-    <>
-      <Button border="none" onClick={() => setIsOpen(!isOpen)}>
+    <Menu
+      onOpen={() => setIsOpen(true)}
+      onClose={() => setIsOpen(false)} // To account for "close by lose focus"
+    >
+      <MenuButton as={Button} border="none">
         <Icon decorative name={IconNames.download} modifiers={['small']} />
         <Text variant="headerNav">Table of Contents</Text>
-      </Button>
+      </MenuButton>
       {isOpen && manifest?.toc && (
-        <ul
-          style={{
-            position: 'fixed',
-            top: `${HEADER_HEIGHT}px`,
-            left: 0,
-            width: '100vw',
-            height: '100vh',
-            background: 'lightgray',
-            padding: '1rem 3rem',
-            margin: 0,
-            overflow: 'auto',
-          }}
-        >
-          {manifest.toc.map((content) => (
-            <li key={content.title}>
-              <Link
-                href={content.href}
-                onClick={tocLinkHandler}
-                fontSize={readerState?.currentTocUrl === content.href ? 3 : 2}
-                _hover={{ fontSize: 3 }}
-              >
-                {content.title}
-              </Link>
-              {content.children && (
-                <ul>
-                  {content.children.map((subLink) => (
-                    <li key={subLink.title} style={{ margin: '15px 25px' }}>
-                      <Link
-                        href={subLink.href}
-                        onClick={tocLinkHandler}
-                        fontSize={
-                          readerState?.currentTocUrl === subLink.href ? 3 : 2
-                        }
-                      >
-                        {subLink.title}
-                      </Link>
-                    </li>
+        <Portal>
+          <MenuList
+            width="100vw"
+            height={`calc(100vh - ${HEADER_HEIGHT}px)`}
+            background={tocBgColor}
+            borderRadius="none"
+            borderColor={tocBgColor}
+            zIndex="popover"
+            px={7}
+            mt="-2px" // Move the popover slightly higher to hide Header border
+            overflow="auto"
+          >
+            {manifest.toc.map((content) => (
+              <React.Fragment key={content.title}>
+                <TocItem
+                  href={content.href}
+                  title={content.title}
+                  isActive={readerState?.currentTocUrl === content.href}
+                  onClick={tocLinkHandler}
```

Comment:
I think what I would do is `onClick={() => handleLinkClick(content.href)}`, but this appears to also work. Just thought I'd leave a comment about it